exec/java-exec/src/main/java/org/apache/drill/exec/physical/base/AbstractBase.java

```diff
@@ -99,17 +100,31 @@ public void setCost(double cost) {
     this.cost = cost;
   }
 
-  // Not available. Presumably because Drill does not currently use
-  // this value, though it does appear in some test physical plans.
-//  public void setMaxAllocation(long alloc) {
-//    maxAllocation = alloc;
-//  }
-
   @Override
   public long getMaxAllocation() {
     return maxAllocation;
   }
 
+  /**
+   * Any operator that supports spilling should override this method
+   * @param maxAllocation The max memory allocation to be set
+   */
+  @Override
+  public void setMaxAllocation(long maxAllocation) {
```

Comment:
That's what happens when you give a software "intelligence" to make decisions for you ..... shall we pick another method name to fool Jackson ?